extensions/ql-vscode/src/remote-queries/variant-analysis-results-manager.ts

```diff
@@ -121,6 +121,7 @@ export class VariantAnalysisResultsManager extends DisposableObject {
       createCacheKey(variantAnalysisId, repositoryFullName),
     );
     if (result) {
+      this._onResultLoaded.fire(result);
```

Comment:
We shouldn't need to do that, that should already be handled in the `loadResultsIntoMemory` method: https://github.com/github/vscode-codeql/blob/bc65d8f31164cf6bb6e2de97450b42b99ac568a2/extensions/ql-vscode/src/remote-queries/variant-analysis-results-manager.ts#L157

Reply:
Yup, but the next if statement is actually calling `loadResultsFromStorage` instead of `loadResultsIntoMemory`:

```
if (options?.skipCacheStore) {
      return this.loadResultsFromStorage(
        variantAnalysisId,
        variantAnalysisStoragePath,
        repositoryFullName,
      );
    }
```

Reply:
That's true, but that if-statement is explicitly created to skip storing the results anywhere, which should include not firing the event. If we did fire the event in there, it would distribute the result to the view, which would then store it in memory. By setting the `skipCacheStore` option, the caller is explicitly saying that they don't want the result to be stored anywhere.

This option is used when exporting results. For example, if there are 1,000 results, you might want to export all 1,000 results to a file. We need to load the SARIF files into memory to create the Markdown files, but this is very temporary. Let's say the SARIF file results in a memory usage of 30MB. Then, we'd use 30MB while exporting, and return to the previous memory usage once exporting is done. If this option is not set, we'd use 30,000 MB (30 GB) after exporting since every result is stored in cache.

If we were to now change the behaviour to fire the event, we would still be storing that 30 GB of memory in the view, rather than in the extension host. I don't think this is what we want to do.

Reply:
I love the example! :) Thanks! That makes sense.